include/cetl/variable_length_array.hpp

```diff
@@ -2404,9 +2420,26 @@ class VariableLengthArray<bool, Allocator> : protected VariableLengthArrayBase<u
         return true;
     }
 
+    constexpr void reserve_bits(const size_type desired_capacity, const size_type max_size)
+    {
+        size_type clamped_capacity = desired_capacity;
+        if (clamped_capacity > max_size)
+        {
+#if defined(__cpp_exceptions)
+            throw std::length_error("Requested capacity exceeds maximum size.");
+#else
+            // deviation from the standard: instead of undefined behaviour we clamp the capacity to the maximum size
+            // when exceptions are disabled.
+            clamped_capacity = max_size;
+#endif
+        }
+        Base::reserve(bits2bytes(clamped_capacity), bits2bytes(max_size));
+    }
+
     constexpr static size_type round8(const size_type value) noexcept
     {
-        return (value + 7U) & ~7U;
+        static_assert(numeric_limits_max_bits() + 7U > numeric_limits_max_bits(), "Uh oh, overflow");
```

Comment:
why is this static_assert here and not in numeric_limits_max_bits()?

Reply:
I put it here because this is where the +7U operation is performed so it seemed like it should go with that.  I could move it to numeric_limits_max_bits() if you want.